File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        ans=0
        mp = {}
        for val in nums:
            complement = k-val
            if complement in mp and mp[complement]:
                ans+=1
                mp[complement]-=1
            else:
                mp[val]= mp.get(val, 0)+1
        return ans       

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        
        unordered_map<int,int>mp;
        const int n= nums.size();
        int ans=0;
        for(int i=0;i<n;i++){
            
            int complement = k- nums[i];
            
            if(mp[complement]>0){
                ans++;
                mp[complement]--;
            }else{
                 mp[nums[i]]++;
            }
        }
        
        return ans;
        
    }
};